cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestTimeSeriesController.java

```diff
@@ -99,6 +103,63 @@ void testRetrieveTimeSeries() throws IOException {
         assertTrue(offset.isEstimate());
     }
 
+    @Test
+    void testRetrieveVersionedTimeSeries() throws IOException {
+        String collect = readJsonFile("radar/v2/json/timeseries_versioned.json");
+        mockHttpServer.enqueue(collect);
+        mockHttpServer.start();
+        Instant start = ZonedDateTime.of(2018, 1, 5, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        Instant end = ZonedDateTime.of(2018, 2, 5, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        Instant versionDate = ZonedDateTime.of(2024, 3, 11, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        TimeSeriesEndpointInput.GetOne input = TimeSeriesEndpointInput.getOne("arbu.Elev.Inst.1Hour.0.Ccp-Rev")
+            .officeId("SWT")
+            .unit("SI")
+            .begin(start)
+            .end(end)
+            .page(null)
+            .versionDate(versionDate);
+        TimeSeries timeSeries = new TimeSeriesController().retrieveTimeSeries(buildConnectionInfo(), input);
+        assertEquals(3, timeSeries.getValues().size());
+        assertEquals(3, timeSeries.getTotal());
+        assertEquals("SWT", timeSeries.getOfficeId());
+        assertEquals("m", timeSeries.getUnits());
+        assertEquals(Duration.ZERO, timeSeries.getInterval());
+        assertEquals(VersionType.SINGLE_VERSION, timeSeries.getVersionType());
+        assertEquals("ARBU.Elev.Inst.1Hour.0.Ccp-Rev", timeSeries.getName());
+        assertEquals(start, timeSeries.getBegin().toInstant());
+        TimeSeriesValues v1 = timeSeries.getValues().get(0);
+        assertEquals(1, v1.getValue(), .001);
+        assertEquals(0, v1.getQualityCode());
+    }
+
+    @Test
+    void testRetrieveMaxAgg() throws IOException {
+        String collect = readJsonFile("radar/v2/json/timeseries_max_agg.json");
+        mockHttpServer.enqueue(collect);
+        mockHttpServer.start();
+        Instant start = ZonedDateTime.of(2018, 1, 5, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        Instant end = ZonedDateTime.of(2018, 2, 5, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        Instant versionDate = ZonedDateTime.of(2024, 3, 11, 0, 0, 0, 0, ZoneId.of("UTC")).toInstant();
+        TimeSeriesEndpointInput.GetOne input = TimeSeriesEndpointInput.getOne("arbu.Elev.Inst.1Hour.0.Ccp-Rev")
+                .officeId("SWT")
+                .unit("SI")
+                .begin(start)
+                .end(end)
+                .page(null);
+        TimeSeries timeSeries = new TimeSeriesController().retrieveTimeSeries(buildConnectionInfo(), input);
+        assertEquals(3, timeSeries.getValues().size());
+        assertEquals(3, timeSeries.getTotal());
+        assertEquals("SWT", timeSeries.getOfficeId());
+        assertEquals("m", timeSeries.getUnits());
+        assertEquals(Duration.ZERO, timeSeries.getInterval());
+        assertEquals(VersionType.MAX_AGGREGATE, timeSeries.getVersionType());
```

Comment:
should also assert that the date is null

Reply:
Fixed